src/Tempest/Core/src/Kernel.php

```diff
@@ -76,29 +81,48 @@ private function createContainer(): Container
         return $container;
     }
 
-    private function loadComposer(): self
+    public function loadComposer(): self
     {
         $this->container->singleton(Composer::class, new Composer($this->root));
 
         return $this;
     }
 
-    private function loadEnv(): self
+    public function loadEnv(): self
     {
+        // The discovery:generate command must always run without discovery cache enabled
```

Comment:
So, this part sucks, but all alternatives are worse.

The `discovery:generate` command must always run with discovery cache disabled. If we don't do that, and switch our discovery caching from `all` to `partial` or the other way around, you end up with a corrupt discovery cache with no way to regenerate it anymore (`./tempest` won't boot).

That's why we also throw an exception a little further in this method, to force people to rerun `discovery:generate` whenever they change their caching strategy (except when you turn discovery cache off).

I considered other places to add this logic, but it _must_ happen before we try to access any dependency from the container, since at that point we need a valid discovery cache.